test/e2e/reconciler/cache/replication_test.go

```diff
@@ -445,8 +548,10 @@ func (b *replicateResourceScenario) getCachedResourceHelper(ctx context.Context,
 		return b.cacheKcpClusterClient.Cluster(cluster).ApisV1alpha1().APIExports().Get(cacheclient.WithShardInContext(ctx, shard.New("root")), b.resourceName, metav1.GetOptions{})
 	case "APIResourceSchema":
 		return b.cacheKcpClusterClient.Cluster(cluster).ApisV1alpha1().APIResourceSchemas().Get(cacheclient.WithShardInContext(ctx, shard.New("root")), b.resourceName, metav1.GetOptions{})
+	case "ClusterWorkspaceShard":
+		return b.cacheKcpClusterClient.Cluster(cluster).TenancyV1alpha1().ClusterWorkspaceShards().Get(cacheclient.WithShardInContext(ctx, shard.New("root")), b.resourceName, metav1.GetOptions{})
```

Comment:
broader question: why are we using strongly typed (as opposed to dynamic) clients and informers in here?

Reply:
leads to more readable code in general and we also make sure that strongly typed clients/informers can be used with the cache server

Reply:
Are we doing special logic for each? Or the same? Convince me that we're not going to gain a case in this switch statement for every type ... ?

Reply:
We won't because caching is expensive :) 

My main goal was to test the kcp client with the cache.
Plus individual tests read well (i.e. https://github.com/kcp-dev/kcp/blob/93e69afd3c82f090282e150ac76eba5a7ffc8dc5/test/e2e/reconciler/cache/replication_test.go#L165). 

If we were to use the dynamic client we would have to use `unstructured`

Reply:
> Also was concerned we're re-writing tons and tons of boilerplate for each type when the logic to handle it is identical

Usually creation and a spec update are different. Initially I captured it with a `baseScenario` idea but Stefan didn't like it (https://github.com/kcp-dev/kcp/pull/2240#discussion_r1012810357).  

Maybe we could have a separate test for testing only the kcp client and then change the replication tests to be more dynamic. I think I could do that, wdyt?

Reply:
Whichever you think is better.

Reply:
okay, i can prepare something and show it you, we can always trash it if we don't like it.

Reply:
this can be done in a separate PR, let's not block this pr on it.